ddtrace/tracer/tracer_test.go

```diff
@@ -1038,11 +1040,71 @@ func TestTracerReportsHostname(t *testing.T) {
 		assert.True(ok)
 		assert.Equal(name, tracer.hostname)
 
+		name, ok = child.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(name, tracer.hostname)
+	})
+
+	t.Run("DD_TRACE_REPORT_HOSTNAME/false", func(t *testing.T) {
+		tracer, _, _, stop := startTestTracer(t)
+		defer stop()
+
+		root := tracer.StartSpan("root").(*span)
+		child := tracer.StartSpan("child", ChildOf(root.Context())).(*span)
+		child.Finish()
+		root.Finish()
+
+		assert := assert.New(t)
+
+		_, ok := root.Meta[keyHostname]
+		assert.False(ok)
 		_, ok = child.Meta[keyHostname]
+		assert.False(ok)
+	})
+
+	t.Run("WithHostname", func(t *testing.T) {
+		tracer, _, _, stop := startTestTracer(t, WithHostname(hostname))
+		defer stop()
+
+		root := tracer.StartSpan("root").(*span)
+		child := tracer.StartSpan("child", ChildOf(root.Context())).(*span)
+		child.Finish()
+		root.Finish()
+
+		assert := assert.New(t)
+
+		got, ok := root.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(got, hostname)
+
+		got, ok = child.Meta[keyHostname]
 		assert.True(ok)
+		assert.Equal(got, hostname)
 	})
 
-	t.Run("disabled", func(t *testing.T) {
+	t.Run("DD_TRACE_SOURCE_HOSTNAME", func(t *testing.T) {
+		os.Setenv("DD_TRACE_SOURCE_HOSTNAME", "hostname-test")
+		defer os.Unsetenv("DD_TRACE_SOURCE_HOSTNAME")
+
+		tracer, _, _, stop := startTestTracer(t)
+		defer stop()
+
+		root := tracer.StartSpan("root").(*span)
+		child := tracer.StartSpan("child", ChildOf(root.Context())).(*span)
+		child.Finish()
+		root.Finish()
+
+		assert := assert.New(t)
+
+		got, ok := root.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(got, hostname)
+
+		got, ok = child.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(got, hostname)
+	})
+	t.Run("DD_TRACE_REPORT_HOSTNAME/nil", func(t *testing.T) {
```

Comment:
Missed an empty line above his. Also, let's just consider it unset, it's not `nil`
```suggestion
	t.Run("DD_TRACE_REPORT_HOSTNAME/unset", func(t *testing.T) {
```

Reply:
Great, I couldn't find a word